src/app/chart/chart.base.ts

```diff
@@ -0,0 +1,45 @@
+import { Input, Output, EventEmitter } from '@angular/core';
+import * as c3 from 'c3';
+import { cloneDeep } from 'lodash';
+import { ChartConfig } from './chart-config';
+
+export abstract class ChartBase {
+
+  /**
+   * Chart configuration object with data
+   */
+  @Input() public config: ChartConfig;
+
+  /**
+   * Event emitted with the chart reference after load is complete
+   * @type {EventEmitter}
+   */
+  @Output() chartLoaded: EventEmitter<any> = new EventEmitter();
+
+  public prevConfig: any;
+  // store the chart object
+  private chart: any;
+
+  constructor() {
+  }
+
+  generateChart (chartId: string, reload?: boolean): void {
```

Comment:
This should probably be protected or private. It's showing up in typedoc without comments.